### ISSUE TITLE
Ball can get left or right the goal (related to more realistic ball trajectory)

### ISSUE THREAD

**Issue by dglaude** [Orange-OpenSource/super-coding-ball#30]
Maybe since the latest change to have a more realistic ball behavior when there is a goal:
https://github.com/Orange-OpenSource/super-coding-ball/commit/6cafd4fc4dc3465665e34b8df5b5225cc62a1fb8

I have seen the ball go too fare on the left or too far on the right (not blocked by "petit fillet").

Here is a capture I could take, but I have seen the ball as far as between the A and the B:
![image](https://github.com/Orange-OpenSource/super-coding-ball/assets/19435932/3701ce03-8ef3-468f-b070-af3ec9923733)

It is hard to remember how it was before, but the new behavior is much better, at least I noticed a change wondering how it was before. 👍

**Comment by dglaude** [comment]
Testing right now... thank you very much in advance.

**Comment by clementcontet** [comment]
Hi, yes it's related to this change.
Before, the ball would "stick" to the goal line, it was not very realistic and had the side effect that the ball could be caught by a defender *after* entering the goal (thus not detecting correctly who had scored the goal)

> Testing right now

Indeed fixed in https://github.com/Orange-OpenSource/super-coding-ball/commit/7a4886cb0d7285228dcc3545f189a230d3dea086

**Comment by dglaude** [comment]
Yes I already noticed goal that were apparently not registered, but I attributed that to being after the mid-time or end of the game. Also difficulties against teams that put many or four players in the goal (but that might be another issue).

But your fix seems perfect, the ball stop at the maximum with the center of the ball on the side net.
